### PR TITLE
App stopped - mobile session finish

### PR DESCRIPTION
**Pull request by jankrzempek**
What was done:

Handeled AppWillTerminate state, so whenever it crash it saved the mobile active session

https://trello.com/c/IkuofDdr/110-if-the-app-has-stopped-crashed-the-mobile-sessions-should-be-finished-on-app-start-check-if-there-are-any-recording-sessions-and

### CODE REVIEW

File: AirCasting/AirCastingApp.swift [hunk 1]
```diff
@@ -8,22 +8,25 @@
...
     private let syncScheduler: SynchronizationScheduler
     private let microphoneManager: MicrophoneManager
+    private let appState: AppState
     private var sessionSynchronizer: SessionSynchronizer
     private let persistenceController = PersistenceController.shared
     private let appBecameActive = PassthroughSubject<Void, Never>()
     @ObservedObject private var offlineMessageViewModel: OfflineMessageViewModel
     private var cancellables: [AnyCancellable] = []
-
+//    let appState = AppState()
```
Comment (smialko) [inline AirCasting/AirCastingApp.swift:24]: Why did you leave this comment? :)

File: AirCasting/AirCastingApp.swift [hunk 2]
```diff
@@ -8,22 +8,25 @@
...
+    
     init() {
-        #if !DEBUG
+//        #if !DEBUG
         FirebaseApp.configure()
-        #endif
+//        #endif
```
Comment (smialko) [inline AirCasting/AirCastingApp.swift:29]: Why did you comment `#if !DEBUG` and `#endif`?

File: AirCasting/AppState.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Created by Lunar on 17/08/2021.
+//
+import SwiftUI
+
+class AppState: ObservableObject {
```
Comment (smialko) [inline AirCasting/AppState.swift:5]: it's only for the microphone sessions while mobile sessions can be both mic or AB :)

File: AirCasting/Settings/MyAccountViewSignOut.swift [hunk 4]
```diff
@@ -32,11 +32,12 @@ private extension MyAccountViewSignOut {
...
-                try logoutController.logout()
-            } catch {
-                assertionFailure("Failed to deauthorize \(error)")
-            }
+            fatalError("CRASHLYTICS TEST")
+//            do {
+//                try logoutController.logout()
+//            } catch {
+//                assertionFailure("Failed to deauthorize \(error)")
+//            }
```
Comment (smialko) [inline AirCasting/Settings/MyAccountViewSignOut.swift:40]: Same as above - commented code, why? :D

File: AirCasting/AirCastingApp.swift [hunk 5]
```diff
@@ -51,6 +54,7 @@ struct AirCastingApp: App {
             RootAppView(sessionSynchronizer: sessionSynchronizer, persistenceController: persistenceController)
                 .environmentObject(authorization)
                 .environmentObject(microphoneManager)
+                .environmentObject(appState)
```
Comment (smialko) [inline AirCasting/AirCastingApp.swift:57]: Why making appState `environmentObject` if you don't use it anywhere?

**Review by smialko — CHANGES_REQUESTED** [review]
Good job, although I have some remarks.
The most important:
1. You missed mobile AB sessions. 
2. You don't cover the "On app start check if there are any recording sessions and finish them if so." part of the task.

File: AirCasting/AirCastingApp.swift (hunk 1 above)
Comment (jankrzempek) [reply AirCasting/AirCastingApp.swift:24]: I cannot see it, maybe from earlier commit? 🙁

File: AirCasting/AirCastingApp.swift (hunk 2 above)
Comment (jankrzempek) [reply AirCasting/AirCastingApp.swift:29]: same: I cannot see it, maybe from earlier commit? 🙁

File: AirCasting/AppState.swift (hunk 3 above)
Comment (jankrzempek) [reply AirCasting/AppState.swift:5]: Yes You are totally right!

File: AirCasting/Settings/MyAccountViewSignOut.swift (hunk 4 above)
Comment (jankrzempek) [reply AirCasting/Settings/MyAccountViewSignOut.swift:40]: same: I cannot see it, maybe from earlier commit? 🙁

File: AirCasting/AirCastingApp.swift (hunk 5 above)
Comment (jankrzempek) [reply AirCasting/AirCastingApp.swift:57]: same: I cannot see it, maybe from earlier commit? 🙁

File: AirCasting/AirCastingApp.swift (hunk 2 above)
Comment (smialko) [reply AirCasting/AirCastingApp.swift:29]: it's weird, why github is marking it as changes if it's different now. Sorry, I can see in the code that it's not commented anymore 🤷🏻‍♀️

File: AirCasting/Settings/MyAccountViewSignOut.swift (hunk 4 above)
Comment (smialko) [reply AirCasting/Settings/MyAccountViewSignOut.swift:40]: Weird, but true, the code is different from what github is suggesting it is.

File: AirCasting/AirCastingApp.swift (hunk 2 above)
Comment (jankrzempek) [reply AirCasting/AirCastingApp.swift:29]: Important that we agreed on this 😛

**Pull request closed without merging**